neo/sys/win32/win_main.cpp

```diff
@@ -602,7 +602,17 @@ uintptr_t Sys_DLL_Load( const char *dllName ) {
 		}
 	} else {
 		DWORD e = GetLastError();
-		// TODO
+		//when encountering errors, pls ammend to list
+		//see https://docs.microsoft.com/en-us/windows/win32/debug/system-error-codes
+		idStr errorStr = va("\n\t\t Unknown Error [%i (0x%X)]",e,e);
+		if (e == 0x7E)
+			errorStr = "\n\t\tERROR_MOD_NOT_FOUND [126 (0x7E)]\n\t\tThe specified module could not be found.";
+		else if (e == 0xD6)
+			errorStr = "\n\t\tERROR_TOO_MANY_MODULES [214 (0xD6)]\n\t\tToo many dynamic - link modules are attached to this program or dynamic - link module.";
+		else if (e == 0xD7)
```

Comment:
are there more errors that are relevant? like 64bit vs 32bit incompatibility?

and wasn't there even a way on windows to get an error string with more information, something with https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-formatmessage ?

Reply:
Im not sure,printing it with formatmessage enough?

Reply:
yes, I hope that formatmessage creates a usable error message.
might be worth trying what happens if you throw a 64bit game dll at a 32bit dhewm3.exe though, if the error message is too cryptic that case could be handled specially (I guess this could happen to people every once in a while if they download mods)

Reply:
WARNING: LoadLibrary(D:/ID4/d3wasm/neo/build/x32-Debug/base.dll) Failed ! [193 (0xC1)] %1 is not a valid Win32 application.

is what it now prints when laoding an 64bit dll in 32bit engine with the arguement parsing. I do not really want to go there tbh (the argument parsing). ><"

Reply:
what do you mean with argument parsing?

Reply:
WARNING: LoadLibrary(D:/ID4/d3wasm/neo/build/x32-Debug/base.dll) Failed ! [193 (0xC1)] **%1** is not a valid Win32 application.

Format message also inserts argument positions (https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-formatmessage) you can jump some hoops to get them filled.

Reply:
Oh - never mind about that.
However, I think it'd make sense to add a special case for `e == 0xC1` and give a more helpful error message, like "[193 (0xC1)] - probably the DLL is of the wrong architecture, like x64 instead of x86".
Also, minor nitpick, I prefer a space after commas in function arguments (`foo(a, b)` instead of `foo(a,b)`) - sorry for being pedantic ;)